src/storage/src/table/cell_based_table.rs

```diff
@@ -51,11 +53,21 @@ pub const READ_WRITE: AccessType = true;
 
 pub const DEFAULT_VNODE: VirtualNode = 0;
 
-/// `CellBasedTable` is the interface accessing relational data in KV(`StateStore`) with encoding
+pub type DedupPkCellBasedTable<S, const T: AccessType> =
+    CellBasedTableBase<S, DedupPkCellBasedRowSerializer, T>;
+
+/// [`CellBasedTable`] is the interface accessing relational data in KV(`StateStore`) with encoding
 /// format: [keyspace | pk | `column_id` (4B)] -> value.
 /// if the key of the column id does not exist, it will be Null in the relation
+pub type CellBasedTable<S, const T: AccessType> = CellBasedTableBase<S, CellBasedRowSerializer, T>;
+
+/// [`CellBasedTableBase`] is the interface accessing relational data in KV(`StateStore`) with
+/// encoding format: [keyspace | pk | `column_id` (4B)] -> value.
+/// if the key of the column id does not exist, it will be Null in the relation.
+/// It is parameterized by its encoding, by specifying cell serializer and deserializers.
+/// TODO: Parameterize on `CellDeserializer`.
 #[derive(Clone)]
-pub struct CellBasedTable<S: StateStore, const T: AccessType> {
+pub struct CellBasedTableBase<S: StateStore, SER: CellSerializer, const T: AccessType> {
```

Comment:
As later we would also `Parameterize on CellDeserializer`, does `DESER: CellDeserializer` and `SER: CellSerializer` have a 1-1 relationship, e.g. not all combinations are valid?

Reply:
> DESER: CellDeserializer and SER: CellSerializer 
I see what you mean. Originally the plan was to parameterize on both of these.
I agree that it would be better to parameterize on `ENC: CellBasedRowEncoding`, which provides matching ser/de interfaces. Will do that instead.